frontend/src/app/pages/marker-page/marker-page.component.ts

```diff
@@ -303,4 +309,58 @@ export class MarkerPageComponent implements OnInit {
                 this.marker.setSliderFocus(true);
             });
     }
+
+    public toggleColorWindowPanel(): void {
+        this.colorWindowPanelActive = !this.colorWindowPanelActive;
+    }
+
+    public resetColorWindowPanel(): void {
+        this.colorWindowWidth = MarkerPageComponent.DEFAULT_COLOR_WINDOW_WIDTH;
+        this.colorWindowCenter = MarkerPageComponent.DEFAULT_COLOR_WINDOW_CENTER;
+        this.rescaleImageColorWindow();
+    }
+
+    public changeImageWindowWidth(newValue: number): void {
+        this.colorWindowWidth = newValue;
+        this.rescaleImageColorWindow();
+    }
+
+    public changeImageWindowCenter(newValue: number): void {
+        this.colorWindowCenter = newValue;
+        this.rescaleImageColorWindow();
+    }
+
+    private rescaleImageColorWindow(): void {
+        this.marker.modifyViewerImage(this.applyWindowScaling.bind(this));
+    }
+
+    private applyWindowScaling(imageRGBBytes: Uint8ClampedArray): Uint8ClampedArray {
+        const lowestVisibleValue = this.colorWindowCenter - (this.colorWindowWidth / 2);
+        const highestVisibleValue = this.colorWindowCenter + (this.colorWindowWidth / 2);
+
+        for (let i = 0; i < imageRGBBytes.length; i += 4) {
+
+            const greyScaleValue = imageRGBBytes[i]; // red channel here, same as the next two beeing greyscale image
+            const alpha = imageRGBBytes[i + 3];
+
+            if (alpha === 0) {
```

Comment:
(Just asking) Is it really possible? If not, maybe it would be better to remove it? From the backend point of view, I think this is impossible. I know that PNG supports alpha channel but we don't use it 😉

Reply:
I'm not sure about it so went with safe way here leaving it. Its just few lines of readable code so let it stay.